docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+
+#### Authentication on applications
+
+Most applications need to know the identity of the user so that they can provide a customized experience.
+
+Applications may start an authentication process, issuing an authentication request to an IDM Wallet, containing the application DID and Verifiable Credentials with its details, such as the application name, and the credentials it wants to receive from an identity. The IDM Wallet will perform DID-Auth to ensure that the application is in control of the DID it presented. If successful, the user is then prompted to select an identity and to accept what the application is asking for.
+
+If the user accepted the prompt, a new session is created and will be used for any further interactions between that application and that IDM Wallet. The session is composed of a key-pair where the public key is known by both, serving as the session identifier, while the private key is only known by that IDM Wallet. Additionally, all sessions have a max-age, meaning they expire as time passes.
+
+#### Signing artifacts on applications
+
+Applications may want to sign artifacts, such as regular data or ephemeral keys. This will be possible in two different ways:
+
+- Sign with the Session Private Key
+- Sign with the Device Private Key
+
+The first method is less intrusive and happens transparently to the user, but is also less secure. More specifically, a robber who steals an IDM Wallet may still sign artifacts with the Session Private Key as long as a session is valid because that key is stored unencrypted. Verifiers will still see those signatures as valid until the DID owner revokes the stolen device.
+
+The second method is more secure as the user is prompted to unlock IDM to decrypt the Device Private Key, but it is more intrusive. The interaction is similar to the authentication process, where the user is prompted inside the IDM Wallet to sign. Because the Device Private Key is encrypted, this gives the DID owner plenty of time to revoke that key in another IDM Wallet.
+
+Ultimately, the application may choose between both methods for different situations depending on the security degree they want to have.
+
+#### Verifying signatures
+
+Any party might verify the authenticity and authorship of artifacts signed by others.
+
+- The verification of an artifact signed by the Session Public Key is performed by:
+    1. Verifying the signature of the artifact against the Session Public Key of the author.
+    2. Verifying the signature of the Session Public Key against the Device Public Key of the author. This proves that the Session Public Key was authorized by the Device Private Key.

```

Comment:
Here you can understand what the device key is used for, but you can not understand why it is used for. Adding the statement I mentioned previously will solve this misunderstanding.